cli/tests/e2e/test_ci.py

```diff
@@ -63,6 +63,34 @@
 FROZEN_ISOTIMESTAMP = "1970-01-01T00:00:00"
 
 
+# To ensure our tests are as accurate as possible, lets try to autodetect what GITHUB_ vars
+# the app code uses, so the tests can enforce the env is mocked appropriately.
+_cli_src = (Path(__file__).parent.parent.parent / "src").resolve()
+USED_GITHUB_VARS = set(
+    subprocess.run(
+        f"git grep -hPo 'GITHUB_[\\w_]*' {_cli_src}", shell=True, capture_output=True
```

Comment:
This currently captures the GITHUB_xxx string in semgrep/interfaces/semgrep_output_v1.atd
which make 'make e2e' fail locally.
It's because I have ```[submodule]
  recurse = true
  fetchJobs = 20
```
in my .gitconfig.

This git grep command seems a bit fragile to differences in environment ...
or maybe we should restrict it to the cli/src/semgrep/app/ directory?

Also what the {_cli_src} mean ?

Reply:
Restricting the grep might the safest, if you're ok with it - then I'd try this first though: https://github.com/returntocorp/semgrep/pull/8818